src/otaclient/ota_core.py

```diff
@@ -701,9 +694,36 @@ def __init__(
 
             self.ca_chains_store = CAChainStore()
 
-        self.started = True
-        logger.info("otaclient started")
-        logger.info(f"firmware_version: {self.current_version}")
+        # load and report booted OTA status
+        _boot_ctrl_loaded_ota_status = self.boot_controller.get_booted_ota_status()
+        if not otaclient_cfg.ECU_INFO_LOADED_SUCCESSFULLY:
+            logger.error(
+                "ecu_info.yaml file is not loaded properly, will reject any OTA request."
+            )
+            logger.error(f"set live_ota_status to {OTAStatus.FAILURE!r}")
+            self._live_ota_status = OTAStatus.FAILURE
+            status_report_queue.put_nowait(
+                StatusReport(
+                    payload=OTAStatusChangeReport(
+                        new_ota_status=OTAStatus.FAILURE,
+                        failure_type=FailureType.UNRECOVERABLE,
+                        failure_reason=f"ecu_info.yaml file is broken or missing, please check {cfg.ECU_INFO_FPATH}. "
+                        "reject any OTA request.",
+                    ),
+                )
+            )
+        else:
+            self._live_ota_status = _boot_ctrl_loaded_ota_status
+            status_report_queue.put_nowait(
+                StatusReport(
+                    payload=OTAStatusChangeReport(
+                        new_ota_status=_boot_ctrl_loaded_ota_status,
+                    ),
+                )
+            )
+
+            self.started = True
+            logger.info("otaclient started")
```

Comment:
If ecu_info.yaml is not loaded properly, we will just set the otaclient as NOT started, any requests coming to otaclient will be rejected when otaclient is not started.

Reply:
Note that when ecu_info.yaml is broken, only the live_ota_status(in the memory) will be set to FAILURE, the `status` file will not be override, as ecu_info.yaml missing/broken is considered a soft failure.